src/app/WriteClient.h

```diff
@@ -105,6 +105,49 @@ class WriteClient : public Messaging::ExchangeDelegate
         virtual void OnDone(WriteClient * apWriteClient) = 0;
     };
 
+    /**
+     *  Initialize the client object. Within the lifetime
+     *  of this instance, this method is invoked once after object
+     *  construction until a call to Shutdown is made to terminate the
+     *  instance.
+     *
+     *  @param[in]    apExchangeMgr    A pointer to the ExchangeManager object.
+     *  @param[in]    apDelegate       InteractionModelDelegate set by application.
+     *  @param[in]    aTimedWriteTimeoutMs If provided, do a timed write using this timeout.
+     *  @retval #CHIP_ERROR_INCORRECT_STATE incorrect state if it is already initialized
+     *  @retval #CHIP_NO_ERROR On success.
+     */
+    CHIP_ERROR Init(Messaging::ExchangeManager * apExchangeMgr, Callback * apDelegate,
```

Comment:
Why do we still have an Init? Should we not follow the ReadClient that just has a constructor?